### PR TITLE
Push v2

### PR DESCRIPTION
**Pull request by marianosz**
Change to support Push on Mobile Apps and Web Apps

### CODE REVIEW

File: sdk/android/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/RegistrationCallback.java [hunk 1]
```diff
@@ -33,5 +33,5 @@
      * @param registration The current registration
```
Comment (soninaren) [inline sdk/android/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/notifications/RegistrationCallback.java]: remove the comment related to registration argument.

File: sdk/android/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/EnhancedPushTests.java [hunk 2]
```diff
@@ -19,225 +19,65 @@
...
-                }
-
-                // create a mock request to replace the existing one
-                ServiceFilterRequestMock requestMock = new ServiceFilterRequestMock(response);
-                return nextServiceFilterCallback.onNext(requestMock);
-            }
-        };
-    }
+    final String appUrl = "http://myapp.com/";
+    final String appKey = "qwerty";
```
Comment (soninaren) [inline sdk/android/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/EnhancedPushTests.java]: remove key

**Comment by soninaren** [comment]
resolving the merge conflicts and sending a new PR

File: sdk/android/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/EnhancedPushTests.java (hunk 2 above)
Comment (marianosz) [inline sdk/android/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/EnhancedPushTests.java]: That key was removed on the Easy Auth branch

**Pull request closed without merging**